src/pages/ProfilesExplorerView/components/SceneProfilesExplorer/SceneProfilesExplorer.tsx

```diff
@@ -304,10 +357,20 @@ export class SceneProfilesExplorer extends SceneObjectBase<SceneProfilesExplorer
   useProfilesExplorer = () => {
     const { explorationType, controls, body, $variables } = this.useState();
 
-    const [timePickerControl, refreshPickerControl] = controls as [SceneObject, SceneObject];
+    const [timePickerControl, refreshPickerControl] =
+      explorationType === ExplorationType.DIFF_FLAME_GRAPH ? [] : (controls as [SceneObject, SceneObject]);
+
     const dataSourceVariable = $variables.state.variables[0] as ProfilesDataSourceVariable;
 
-    const { variables: sceneVariables, gridControls } = (body?.state.primary as any).getVariablesAndGridControls() as {
+    const bodySceneObject = body?.state.primary as any;
+
+    if (typeof bodySceneObject.getVariablesAndGridControls !== 'function') {
+      throw new Error(
```

Comment:
We should use TypeScript to enforce this in the future (assuming we'll keep the same layout with the Scene controls being sticky in the UI).

Reply:
Unless the code is modified, it will not break in runtime. 
If so, the E2E smoke tests would catch it directly. Which makes me think that... we should add a new one for the diff view!